tests/unit/test_credentials.py

```diff
@@ -129,15 +152,23 @@ async def test_ec2_instance_metdata_credentials(
     assert await imc.get_key(http) == metadata.key
 
 
+@pytest.mark.parametrize("use_v2", [True, False])
 async def test_simultaneous_credentials_refresh(
-    http: HttpImplementation, instance_metadata_server: InstanceMetadataServer
+    http: HttpImplementation,
+    instance_metadata_server: InstanceMetadataServer,
+    use_v2: bool,
 ) -> None:
+    model = (
+        InstanceMetadataCredentialsWithImdsV2
+        if use_v2
+        else InstanceMetadataCredentialsWithImdsV1
+    )
```

Comment:
why not parametrize model using `parametrize("model", [InstanceMetadataCredentialsWithImdsV2, InstanceMetadataCredentialsWithImdsV1])`?